lib/project-view.coffee

```diff
@@ -124,8 +138,10 @@ module.exports = ProjectView =
     # Shorten root path if possible
     userHome = fs.getHomeDirectory()
     normRootPath = path.normalize(rootPath)
-    if normRootPath.indexOf(userHome) is 0
-      # Use also tilde in case of Windows as synonym for the home folder
+    if atom.config.get 'project-view.regexMatch'
```

Comment:
They seems same. new RegExp("") equals /(?:)/.
If setting regex to non-empty if view.regexMatch isn't an empty string, then it must be cleared when  regexMatch is cleared, which adds a lot code for this simple logic.